integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -566,6 +569,47 @@ class CarbonScanRDD[T: ClassTag](
           }
           havePair = false
           val value = reader.getCurrentValue
+          if (CarbonProperties.getInstance()
+                .getProperty(CarbonCommonConstants.CARBON_SPARK_VERSION_SPARK3,
+                  CarbonCommonConstants.CARBON_SPARK_VERSION_SPARK3_DEFAULT)
+                .toBoolean &&
+              model.getProjectionColumns.exists(_.getDataType == DataTypes.TIMESTAMP)) {
+            value match {
+              case row: GenericInternalRow if
+                reader.isInstanceOf[CarbonRecordReader[T]] &&
+                null !=
+                reader.asInstanceOf[CarbonRecordReader[T]].getCarbonDataFileWrittenVersion &&
+                // carbonDataFileWrittenVersion will be in the format x.x.x-SNAPSHOT
+                // (eg., 2.1.0-SNAPSHOT), get the version name and check if the data file is
+                // written before 2.2.0 version, then rebase timestamp value
```

Comment:
u can add this comment after line 577 and then correct the style of if condition code